server/form-pages/apply/risk-and-need-factors/oasys-import/roshSummary.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable no-underscore-dangle */
+import type { DataServices, PersonRisksUI } from '@approved-premises/ui'
+
+import type {
+  Application,
+  ArrayOfOASysRiskOfSeriousHarmSummaryQuestions,
+  OASysQuestion,
+  OASysSections,
+} from '@approved-premises/api'
+
+import TasklistPage from '../../../tasklistPage'
+
+import { Page } from '../../../utils/decorators'
+
+type RoshSummaryBody = {
+  roshAnswers: Array<string> | Record<string, string>
+  roshSummaries: ArrayOfOASysRiskOfSeriousHarmSummaryQuestions
+}
+
+@Page({
+  name: 'rosh-summary',
+  bodyProperties: ['roshAnswers', 'roshSummaries'],
+})
+export default class RoshSummary implements TasklistPage {
+  title = 'Edit risk information'
+
+  roshSummary: ArrayOfOASysRiskOfSeriousHarmSummaryQuestions
+
+  risks: PersonRisksUI
+
+  roshAnswers: Record<string, string>
+
+  constructor(public body: Partial<RoshSummaryBody>) {}
+
+  static async initialize(
+    body: Record<string, unknown>,
+    application: Application,
+    token: string,
+    dataServices: DataServices,
+  ) {
+    const oasysSections: OASysSections = await dataServices.personService.getOasysSections(
+      token,
+      application.person.crn,
+    )
+    const risks = await dataServices.personService.getPersonRisks(token, application.person.crn)
+
+    const roshSummary = oasysSections.roshSummary.sort((a, b) => Number(a.questionNumber) - Number(b.questionNumber))
+    const roshAnswers = body?.roshAnswers ? body.roshAnswers : []
+
+    body.roshSummaries = ((roshSummary || []) as Array<OASysQuestion>).map((summary, i) => {
+      return {
+        ...summary,
+        answer: roshAnswers[i],
+      }
+    })
+
+    const page = new RoshSummary(body)
+    page.roshSummary = roshSummary
+    page.risks = risks
+
+    return page
+  }
+
+  previous() {
+    return 'optional-oasys-sections'
+  }
+
+  next() {
+    return ''
+  }
+
+  roshTextAreas() {
```

Comment:
I'm trying to get us to move some of this presentation layer stuff to helpers to be used in views. This can probably go in as is, as this looks like something that we'll be able to share across the other OAsys pages 😃

Reply:
Yeah I can extract the whole function to a helper in the next PR